src/MDMTimingCurve.h

```diff
@@ -0,0 +1,152 @@
+/*
+ Copyright 2017-present The Material Motion Authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <CoreGraphics/CoreGraphics.h>
+#import <Foundation/Foundation.h>
+#import <QuartzCore/QuartzCore.h>
+
+/**
+ The possible kinds of traits curves that can be used to describe an animation.
+ */
+typedef NS_ENUM(NSUInteger, MDMTimingCurveType) {
+  /**
+   The value will be instantly set with no animation.
+   */
+  MDMTimingCurveTypeInstant,
+
+  /**
+   The value will be animated using a cubic bezier curve to model its velocity.
+   */
+  MDMTimingCurveTypeBezier,
+
+  /**
+   The value will be animated using a spring simulation.
+
+   A spring will treat the duration property of the traits as a suggestion and may choose to
+   ignore it altogether.
+   */
+  MDMTimingCurveTypeSpring,
+
+} NS_SWIFT_NAME(TimingCurveType);
+
+/**
+ A generalized representation of a traits curve.
+ */
+typedef struct MDMTimingCurve {
+  /**
+   The type defines how to interpret the data values.
+   */
+  MDMTimingCurveType type;
+
+  /**
+   The data values corresponding with this curve.
+   */
+  CGFloat data[4];
+} NS_SWIFT_NAME(TimingCurve) MDMTimingCurve;
+
+/**
+ Creates a bezier traits curve with the provided control points.
+
+ A cubic bezier has four control points in total. We assume that the first control point is 0, 0 and
+ the last control point is 1, 1. This method requires that you provide the second and third control
+ points.
+
+ See the documentation for CAMediaTimingFunction for more information.
+ */
+// clang-format off
+FOUNDATION_EXTERN
+MDMTimingCurve MDMTimingCurveMakeBezier(CGFloat p1x, CGFloat p1y, CGFloat p2x, CGFloat p2y)
```

Comment:
Why not a pair of CGPoint? Why 4 explicit CGFloat? It seems that if they represent 2 points, it's best to use CGPoint to provide additional context.  Also consider naming "p2x, p2y, p3x, p3y" to match the method documentation.

Reply:
This is in keeping consistent with the CAMediaTimingFunction API. Will change the names.

Reply:
Ack.